django/db/models/lookups.py

```diff
@@ -362,7 +362,7 @@ def process_rhs(self, compiler, connection):
 class IsNull(BuiltinLookup):
     lookup_name = 'isnull'
 
-    def as_sql(self, compiler, connection):
+    def as_sql(self, compiler, connection, **kwargs):
         sql, params = compiler.compile(self.lhs)
```

Comment:
At this position even ``connection`` is not used either. But it remains there because it is the ``as_sql``-function signature. I think this counts also for ``**kwargs``.

The benefit of ``**kwargs`` is also where super-methods are called and use ``**kwargs``. If in the future the super-method makes use of more kwargs, then the child-methods do not have to be adapted. Just a thought, maybe I'm wrong here.

The other consistent choice would be to remove ``**kwargs`` where it is not used in the method or through a call to a super-method that makes use of it.
Even further this could be changed from ``**kwargs`` to the single kwargs like ``template``, ``function`` or ``arg_joiner``.
This would not change the purpose of this ticket: Do not touch ``self`` in ``as_vendor``.

I'm OK with the new function-signature but will do any further changes if needed.